runtime/android/core_internal/src/org/xwalk/core/internal/XWalkViewDelegate.java

```diff
@@ -145,13 +153,73 @@ private static void init(final Context context) {
             CommandLine.init(readCommandLine(context.getApplicationContext()));
         }
 
-        ResourceExtractor.setMandatoryPaksToExtract(MANDATORY_PAKS);
-        int resListResId = context.getResources().getIdentifier(
+        final boolean isSharedMode = !context.getPackageName().equals(
+                context.getApplicationContext().getPackageName());
+
+        HashMap<String, ResourceEntry> resourceList = new HashMap<String, ResourceEntry>();
+        if (isSharedMode) {
+            for (String resource : MANDATORY_PAKS) {
+                resourceList.put(resource, new ResourceEntry(0, "", resource));
+            }
+        } else {
+            int resListId = context.getResources().getIdentifier(
                 XWALK_RESOURCES_LIST_RES_NAME, "array", context.getClass().getPackage().getName());
-        if (resListResId == 0) {
-            resListResId = context.getResources().getIdentifier(
+            if (resListId == 0) {
+                resListId = context.getResources().getIdentifier(
                     XWALK_RESOURCES_LIST_RES_NAME, "array", context.getPackageName());
+            }
+
+            try {
```

Comment:
Then this commit needs more fixes. As I said, this assertion _is_ being hit whenever you launch one of the embedded shell APKs we build for testing, whereas this did not happen before. I can only merge this PR without this regression.